### ISSUE TITLE
speed comparison of dplyr and tidierdata

### ISSUE THREAD

**Issue by Zhaoju-Deng** [TidierOrg/Tidier.jl#116]
Hi Karandeep,
Its nice to have the tidyverse package in Julia!
I tried TidierData to create two new columns for a ~1Gb dataset, I was actually expecting TidierData much more faster than dplyr, but the results showed dplyr was much faster than TidierData (0.9s in dplyr and 5.5-8s in TidierData), would it possible to fine-tune the speed  so that TidierData would be much more efficient in manipulating large datasets. Personally I think that matters for data analysis.

kind regards,
Zhaoju

**Comment by kdpsingh** [comment]
Benchmarking is a tricky issue in Julia because the first time you run code in Julia, it is compiled (leading to a compilation delay). The code usually runs much, much faster after code is compiled. This is true even if you change the underlying dataset or certain parameters, so it's not that Julia is cheating from having cached the answer but is legimately way faster on the second run.

This issue is been mitigated in Julia 1.9 by precompilation, which caches compiled code. DataFrames.jl (which TidierData.jl wraps) takes advantage of precompilation workflows, but TidierData.jl for now doesn't do any additional precompilation (which we probably should).

I'll move this issue to the TidierData.jl repo, because I agree we should periodically run some basic benchmarks to understand how TidierData.jl stacks up against DataFrames.jl (to understand how much overhead is added) and against R tidyverse.

tl;dr: I agree speed is important. Many published benchmarks show DataFrames.jl to be faster than R tidyverse, but we on the first run the compilation can introduce a delay.

Two questions:
- which version of Julia are you running?
- can you try running your Julia code two times and see if the code is faster on the second run?
